packager/signer/signer.go

```diff
@@ -464,15 +464,22 @@ func formatLinkHeader(preloads []*rpb.Metadata_Preload) (string, error) {
 	return strings.Join(values, ","), nil
 }
 
-// serveSignedExchange does the actual work of transforming, packaging and signed and writing to the response.
-func (this *Signer) serveSignedExchange(resp http.ResponseWriter, fetchResp *http.Response, signURL *url.URL, act string, transformVersion int64) {
+// serveSignedExchange_ConsumeCapCompliantFetchResp method consumes the gateway
+// response body, and checks if it's shorter than maxBodyLength and is therefore
+// OK to sign. It does so keeping memory consumption below maxBodyLength. It
+// returns ok iff succeeded and handles the errors if any. Specifically:
```

Comment:
I'd reword this sentence a bit. Something like:
"If successful, it returns true and stores the result in fetchBody. Otherwise, it returns false and writes a response to resp; the caller should not write to resp in this case."